src/par_iter/mod.rs

```diff
@@ -575,6 +576,28 @@ pub trait ParallelIterator: Sized {
         find::find(self, predicate)
     }
 
+    /// Searches for the **first** item in the parallel iterator that
+    /// matches the given predicate and returns it.
+    ///
+    /// Once a match is found, all attempts to the right of the match
+    /// will be stopped, while attempts to the left must continue in case
+    /// an earlier match is found.
+    fn find_first<FIND_OP>(self, predicate: FIND_OP) -> Option<Self::Item>
+        where FIND_OP: Fn(&Self::Item) -> bool + Sync {
+        find_first_last::find_first(self, predicate)
+    }
+
+    /// Searches for the **last** item in the parallel iterator that
+    /// matches the given predicate and returns it.
+    ///
+    /// Once a match is found, all attempts to the left of the match
+    /// will be stopped, while attempts to the right must continue in case
+    /// a later match is found.
+    fn find_last<FIND_OP>(self, predicate: FIND_OP) -> Option<Self::Item>
+        where FIND_OP: Fn(&Self::Item) -> bool + Sync {
+        find_first_last::find_last(self, predicate)
+    }
+
     #[doc(hidden)]
     #[deprecated(note = "parallel `find` does not search in order -- use `find_any`")]
```

Comment:
let's update this deprecated note to say: "use `find_any`, `find_first`, or `find_last`" -- we might also consider calling `find_first()`, not sure.